LoopKitUI/Views/TherapySettingsView.swift

```diff
@@ -0,0 +1,342 @@
+//
+//  TherapySettingsView.swift
+//  LoopKitUI
+//
+//  Created by Rick Pasetto on 7/7/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import AVFoundation
+import HealthKit
+import LoopKit
+import SwiftUI
+
+public protocol TherapySettingsViewDelegate: class {
+    func gotoEdit(therapySetting: TherapySetting)
+    func save()
+    func cancel()
+}
+
+public class TherapySettingsViewModel: ObservableObject {
+    private var initialTherapySettings: TherapySettings
+    var therapySettings: TherapySettings
+
+    public init(therapySettings: TherapySettings) {
+        self.therapySettings = therapySettings
+        self.initialTherapySettings = therapySettings
+    }
+    
+    /// Reset to original
+    func reset() {
+        therapySettings = initialTherapySettings
+    }
+}
+
+public struct TherapySettingsView: View, HorizontalSizeClassOverride {
+    @Environment(\.dismiss) var dismiss
+
+    weak var delegate: TherapySettingsViewDelegate?
+    
+    @ObservedObject var viewModel: TherapySettingsViewModel
+
+    @State var isEditing: Bool = false
+    
+    // Hack to test edit button below section or next to header
+    public enum DesignVersion {
+        case A, B, C
+        func cycle() -> DesignVersion {
+            switch self {
+            case .A: return .B
+            case .B: return .C
+            case .C: return .A
+            }
+        }
+    }
```

Comment:
I offered.  I will rip it out once they decide.  The reason I did it was because I wasn't sure how easy each was, but implementing them was easier than I thought, so I decided to give them an ad-hoc build with this "feature" of switching.